### ISSUE TITLE
asteriskLogger fails without reasonable message when internet connection is unavailable

### ISSUE THREAD

**Issue by ghost** [blak3r/yaai#39]
asteriskLogger needs to access the internet in order for it's SOAP client to work. But this isn't made clear anywhere in the documentation nor in the output:

```
******** Asterisk Logger Starting **************
# Sugar root set to [/srv/http/sugar/6_5_2_yaai24/custom/modules/Asterisk/../../../]
Logging to: /tmp/yaai.log
# Callout prefix is []
# Callin prefix is []
asteriskMatchInternal = /^(sip\/[1-9][0-9][0-9]?[0-9]?-|Local)/i
# (Config processed)
Selecting DB Name: SUGAR_DB_6_5_2_yaai24
! Trying SOAP login endpoint=[http://192.168.12.21:81/sugar/6_5_2_yaai24/soap.php] user=[admin] password=[0cc175b9c0f1b6a831c399e269772661]
PHP Warning:  Attempt to modify property of non-object in /srv/http/sugar/6_5_2_yaai24/include/nusoap/nusoap.php on line 6163
Warning: Attempt to modify property of non-object in /srv/http/sugar/6_5_2_yaai24/include/nusoap/nusoap.php on line 6163
PHP Warning:  Attempt to modify property of non-object in /srv/http/sugar/6_5_2_yaai24/include/nusoap/nusoap.php on line 6163
Warning: Attempt to modify property of non-object in /srv/http/sugar/6_5_2_yaai24/include/nusoap/nusoap.php on line 6163
PHP Warning:  Attempt to modify property of non-object in /srv/http/sugar/6_5_2_yaai24/include/nusoap/nusoap.php on line 6163
Warning: Attempt to modify property of non-object in /srv/http/sugar/6_5_2_yaai24/include/nusoap/nusoap.php on line 6163
PHP Warning:  Attempt to modify property of non-object in /srv/http/sugar/6_5_2_yaai24/include/nusoap/nusoap.php on line 6163
Warning: Attempt to modify property of non-object in /srv/http/sugar/6_5_2_yaai24/include/nusoap/nusoap.php on line 6163
! FATAL: SOAP login failed, something didnt get set by login... check your site_url: user=admin GUID=
```

Only after a few hours digging in meaningless backtraces did I find nusoap had a debug output and hacked it to write to stderr:

```
[...]
2012-07-24 08:47:36.686177 SugarSoap: checkWSDL
2012-07-24 08:47:36.686513 SugarSoap: got wsdl error: Getting http://schemas.xmlsoap.org/wsdl/ - HTTP ERROR: Couldn't open socket connection to server http://schemas.xmlsoap.org/wsdl/ prior to connect().  This is often a problem looking up the host name.
2012-07-24 08:47:36.686854 SugarSoap: call: operation=get_user_id, namespace=http://tempuri.org, soapAction=, rpcParams=, style=rpc, use=encoded, endpointType=wsdl
2012-07-24 08:47:36.686918 SugarSoap: No data for operation: get_user_id
2012-07-24 08:47:36.687122 SugarSoap: operation 'get_user_id' not present in WSDL.
! FATAL: SOAP login failed, something didnt get set by login... check your site_url: user=admin GUID=
```

As the error comes from nusoap, it can't probably be fixed in yaai; however, this issue might be worth mentioning in the docs.

**Comment by blak3r** [comment]
I'm surprised that it does.  Thanks for pointing this out. 

You were testing with the local IP address and not a host name?  What requires Internet?

~blake

**Comment by ghost** [comment]
I was testing it with a local IP, that is OK. The SOAP library requires internet for its internal needs, i.e. to fetch http://schemas.xmlsoap.org/wsdl/, as can be seen from the log. That has nothing to do with the location of my Sugar instance -- that's why it's confusing. It's unpleasant that it has to fetch such a thing at all (there is no principial reason that connecting to a local SOAP server should require fetching something from the internet), but at least it's better to know about it.

**Comment by blak3r** [comment]
You're right.  NuSOAP downloads the schema definitions from the internet.

I suspect we could save the schema locally and specify the $namespaces as shown here:
http://stackoverflow.com/questions/4832415/how-to-disable-external-schema-downloads-in-nusoap

I'm hesitant to get too involved with that right now as someone else already rewrote asteriskLogger to use native soap support and that would also get rid of those error messages in AsteriskLogger which confuse lots of people...

**Comment by gl1tchh** [comment]
Hi,

Does anyone know of a definitive resolution to this issue yet? Having a problem with an installation, I can solve it using hardware but if there is a way around it through configuration that'd be fantastic.

**Comment by blak3r** [comment]
How can you solve it is ring "hardware"?

~blake

**Comment by gl1tchh** [comment]
Well the site of the installation in question has 2 routers, one with a DSL that only allows voice traffic (which is what the server is currently connected with, and what is causing the problem), and the other with a standard DSL line (which I can't use to connect the server, because of voice quality issues). I'll put in a small managed switch that routes all the http traffic from the server over the standard DSL, while keeping voice traffic routed over the dedicated line. So not really a solution, more of a workaround.

**Comment by blak3r** [comment]
Is your sugar instance on the same box as the asterisk server. Only the box running asterisk logger would need the network connection. 

~blake

**Comment by camartinezcha** [comment]
Hi! black3r :+1: 

I need help with SugarCRM SOAP Connection, I have output in my server internet but always get the following error:

Trying SOAP login endpoint=[http://10.5.0.240/sugar/soap.php] user=[admin] password=[21232f297a57a5a743894a0e4a801fc3]
  **FATAL: SOAP login failed, something didn't get set by login... check your site_url, and make sure sugarcrm is running.  SessionID: user=admin GUID=**
  Sleeping for 5 seconds then retrying...

help please! :(

**Comment by MazZzaka** [comment]
try with older php version...

**Comment by eggiecutie** [comment]
try checking if your username and password for your soap is set correctly on the callinize configuration module
